pandas/tests/series/test_constructors.py

```diff
@@ -1464,3 +1464,9 @@ def test_constructor_sparse_datetime64(self, values):
         arr = pd.arrays.SparseArray(values, dtype=dtype)
         expected = pd.Series(arr)
         tm.assert_series_equal(result, expected)
+
+    def test_construction_from_ordered_set(self):
+        # https://github.com/pandas-dev/pandas/issues/36044
+        result = Series({"a": 1, "b": 2}.keys())
```

Comment:
should we get the same behavior with dict_values?

Reply:
Good call, this is actually broken for dict_values. Any thoughts on how best to check for this? I only need to add it to the isinstance above.

Reply:
https://stackoverflow.com/questions/43920587/what-is-the-explicit-python3-type-for-dict-keys-for-isinstance-check